builtin/providers/cobbler/resource_cobbler_profile.go

```diff
@@ -0,0 +1,370 @@
+	d.Set("virt_ram", profile.VirtRam)
+	d.Set("virt_type", profile.VirtType)
+
+	return nil
+}
+
+func resourceProfileUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Create a cobblerclient.Profile struct
+	profile := buildProfile(d, config)
+
+	// Attempt to update the profile with new information
+	log.Printf("[DEBUG] Cobbler Profile: Updating Profile (%s) with options: %+v", d.Id(), profile)
+	err := config.cobblerClient.UpdateProfile(&profile)
+	if err != nil {
+		return fmt.Errorf("Cobbler Profile: Error Updating (%s): %s", d.Id(), err)
+	}
+
+	return resourceProfileRead(d, meta)
+}
+
+func resourceProfileDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	// Attempt to delete the profile
+	if err := config.cobblerClient.DeleteProfile(d.Id()); err != nil {
+		return fmt.Errorf("Cobbler Profile: Error Deleting (%s): %s", d.Id(), err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// buildProfile builds a cobblerclient.Profile out of the Terraform attributes
+func buildProfile(d *schema.ResourceData, meta interface{}) cobbler.Profile {
+	mgmtClasses := []string{}
+	for _, i := range d.Get("mgmt_classes").([]interface{}) {
+		mgmtClasses = append(mgmtClasses, i.(string))
+	}
+
+	nameServersSearch := []string{}
+	for _, i := range d.Get("name_servers_search").([]interface{}) {
+		nameServersSearch = append(nameServersSearch, i.(string))
+	}
+
+	nameServers := []string{}
+	for _, i := range d.Get("name_servers").([]interface{}) {
+		nameServers = append(nameServers, i.(string))
+	}
+
+	owners := []string{}
+	for _, i := range d.Get("owners").([]interface{}) {
+		owners = append(owners, i.(string))
+	}
+
+	repos := []string{}
+	for _, i := range d.Get("repos").([]interface{}) {
+		repos = append(repos, i.(string))
+	}
+
+	profile := cobbler.Profile{
+		BootFiles:              d.Get("boot_files").(string),
+		Comment:                d.Get("comment").(string),
+		Distro:                 d.Get("distro").(string),
+		EnableGPXE:             d.Get("enable_gpxe").(bool),
+		EnableMenu:             d.Get("enable_menu").(bool),
+		FetchableFiles:         d.Get("fetchable_files").(string),
+		KernelOptions:          d.Get("kernel_options").(string),
+		KernelOptionsPost:      d.Get("kernel_options_post").(string),
+		Kickstart:              d.Get("kickstart").(string),
+		KSMeta:                 d.Get("ks_meta").(string),
+		MGMTClasses:            mgmtClasses,
+		MGMTParameters:         d.Get("mgmt_parameters").(string),
+		Name:                   d.Get("name").(string),
+		NameServersSearch:      nameServersSearch,
+		NameServers:            nameServers,
+		Owners:                 owners,
+		Parent:                 d.Get("parent").(string),
+		Proxy:                  d.Get("proxy").(string),
+		RedHatManagementKey:    d.Get("redhat_management_key").(string),
+		RedHatManagementServer: d.Get("redhat_management_server").(string),
+		Repos:                    repos,
```

Comment:
It's a quirk with `go fmt`... I tried realigning, but it just reverted. I think there's one or two other areas where this happens.

Should I leave as-is or look for a way to resolve?